### PR TITLE
feat(wallet): Report Warnings

### PR DESCRIPTION
**Pull request by coreyphillips**
This PR:
- Updates `TStorageWarning` object to include `warningId` and `warningReported` status.
- Attempts to report warnings immediately after detection.
- Updates `defaultChecksShape`.
- Attempts to report any unreported warnings on each start.

Notes:
- We're still waiting on the correct url for reporting warnings, but wanted to create the PR to start review in the meantime.
- Once the endpoint is up, I'll update the fetch request in the `reportImpactedAddressBalance` method, uncomment the code snippet and remove the `TODO`.

### CODE REVIEW

**Comment by limpbrains** [comment]
Type check is failing

**Comment by coreyphillips** [comment]
@pwltr any idea what's triggering the type-check to fail?

Edit: nvm, I believe I found it.

**Review by limpbrains — APPROVED** [review]

File: src/components/List.tsx [hunk 1]
```diff
@@ -55,7 +55,7 @@ type TItemDraggable = {
 export type ItemData = SwitchItem | ButtonItem | TextButtonItem | DraggableItem;
 
 export type SwitchItem = {
-	type: EItemType.switch;
+	type: EItemType;
```
Comment (pwltr) [inline src/components/List.tsx]: Avoid changing this type if possible, see other suggestion below to fix the type error.

```suggestion
	type: EItemType.switch;
```

File: src/screens/Settings/DevSettings/index.tsx [hunk 2]
```diff
@@ -18,156 +19,209 @@ import { resetFeesStore } from '../../../store/actions/fees';
...
+		},
+		{
+			title: 'Wallet Checks',
+			data: [
+				{
+					title: `Warnings: ${warnings.length}`,
+					type: EItemType.textButton,
+					testID: 'Warnings',
+					onPress: (): void => {},
+				},
```
Comment (pwltr) [inline src/screens/Settings/DevSettings/index.tsx]:
```suggestion
				{
					title: `Warnings: ${warnings.length}`,
					type: EItemType.textButton,
					value: '',
					testID: 'Warnings',
				},
```

**Review by pwltr — CHANGES_REQUESTED** [review]

**Review by pwltr — APPROVED** [review]

**Pull request merged**